src/tensors/vectors.jl

```diff
@@ -0,0 +1,23 @@
+struct OffByOneVector{T} <: AbstractVector{T}
```

Comment:
Could we add some docs about this to the manual instead of talking about CIndex?

Reply:
I don't think any code that relies on CIndices will break if we remove it from Finch, since CIndices is an external package. There's some commented code that users it in binsparse.jl which could be replaced with this vector, and a test that uses them where we could also add the solutions developed here. I think we should also simplify the docs to avoid the unsafe pointer stuff.

Reply:
We could leave the docs, but add a warning that the approach is no longer recommended

Reply:
We would really only use it when we need to interact with julia code that doesn't let us modify the vector type. We might consider deferring to the docs CIndex already has, or moving our docs to CIndex.jl

Reply:
Sure! I removed `CIndex` from the test and `binsparse.jl`, and added a note to `CIndex` docs.

Reply:
https://documenter.juliadocs.org/stable/showcase/#Admonitions

Reply:
I would use a warning admonition

Reply:
Done! I added `!!! warning` admonition.